events/nats_publish.go

```diff
@@ -0,0 +1,204 @@
+package events
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+
+	"go.infratographer.com/x/echojwtx"
+	"go.infratographer.com/x/gidx"
+)
+
+// Publish publishes the provided message to the provided topic.
+func (c *NatsConnection) Publish(ctx context.Context, topic string, message any) (Message[any], error) {
+	_, span := newEventTraceSpan(ctx, c.tracer, "events.nats.Publish", topic, message)
+
+	defer span.End()
+
+	// TODO: maybe switch to using nats headers instead of injecting in the message?
```

Comment:
I think leaving it in the message is good for now. I know we want to be able to support more than just NATS in the future and don't want to make any assumptions about what we can do with headers / have to change logic for where to grab things based upon what broker we're using

Reply:
Publish was removed as there was no need yet. Resolving this issue.